cmd/errors/errors.go

```diff
@@ -0,0 +1,26 @@
+package retryerr
+
+import (
+	"errors"
+	"net"
+	"syscall"
+
+	"github.com/jackc/pgerrcode"
+	"github.com/lib/pq"
+)
+
+func CheckErrorType(err error) bool {
+	if err, ok := err.(*pq.Error); ok {
+		if (err.Code == pgerrcode.ConnectionException) || (err.Code == pgerrcode.ConnectionDoesNotExist) || (err.Code == pgerrcode.ConnectionFailure) || (err.Code == pgerrcode.InvalidTransactionInitiation) {
+			return true
+		}
+	}
+
+	if net, ok := err.(net.Error); ok {
+		if errors.Is(net, syscall.ECONNREFUSED) || errors.Is(net, syscall.ETIMEDOUT) || errors.Is(net, syscall.EADDRNOTAVAIL) || errors.Is(net, syscall.EHOSTUNREACH) {
+			return true
+		}
+	}
+
```

Comment:
Почему-то не ловит вот такое:
```
Error while sending metrics: Post "http://localhost:8080/updates/": dial tcp [::1]:8080: connectex: No connection could be made because the target machine actively refused it.
```
По крайней мере, при выключенном сервере я ожидал получить три попытки отправки метрик от ретраера, но получил только одну.